spec/javascripts/angular/controllers/dashboard_ctrl_spec.js

```diff
@@ -0,0 +1,55 @@
+describe('DashboardCtrl', function() {
+  var $controller;
+  var $scope;
+  beforeEach(inject(function(_$controller_, $rootScope){
+    $controller = _$controller_;
+    $scope = $rootScope.$new(true);
+  }));
+
+  describe('visual toggles', function() {
+    describe('clone modal', function() {
+      beforeEach(function() {
+        $controller('DashboardCtrl', { $scope: $scope });
+      });
+
+      it('starts closed', function() {
+        expect($scope.showCloneControls).toEqual(false);
+      });
+
+      it('opens the controls', function() {
+        $scope.showCloneMenu();
+        expect($scope.showCloneControls).toEqual(true);
+      });
+
+      it('can close the modal', function() {
+        expect($scope.showCloneControls).toEqual(false);
+        $scope.showCloneMenu();
+        expect($scope.showCloneControls).toEqual(true);
+        $scope.closeCloneControls();
+        expect($scope.showCloneControls).toEqual(false);
+      });
+    });
+
+    describe('fullscreen', function() {
+      beforeEach(function() {
+        $controller('DashboardCtrl', { $scope: $scope });
+      });
+
+      it('starts not in fullscreen', function() {
+        expect($scope.fullscreen).toEqual(false);
+      });
+
+      it('toggles to fullscreen', function() {
+        $scope.enableFullscreen();
+        expect($scope.fullscreen).toEqual(true);
```

Comment:
right. these are just unit tests, so they don't hook into the visuals.

we have two options for feature specs:

we can stick with capybara, or we can move to protractor (a node.js based solution). protractor gives finer control over checking the DOM (you can see in some of the tests for stz), but it's an additional dependency and I'm not sure how to interface between the protractor being run by node and ruby-side things (setting up servers and dashboards).